bazel/swift_static_framework.bzl

```diff
@@ -0,0 +1,190 @@
+"""
+This rules creates a fat static framework that can be included later with
+static_framework_import
+"""
+
+load("@build_bazel_apple_support//lib:apple_support.bzl", "apple_support")
+load("@build_bazel_rules_swift//swift:swift.bzl", "SwiftInfo", "swift_library")
+
+MINIMUM_IOS_VERSION = "10.0"
+
+_PLATFORM_TO_SWIFTMODULE = {
+    "ios_armv7": "arm",
+    "ios_arm64": "arm64",
+    "ios_i386": "i386",
+    "ios_x86_64": "x86_64",
+}
+
+def _zip_binary_arg(module_name, input_file):
+    return "{module_name}.framework/{module_name}={file_path}".format(
+        module_name = module_name,
+        file_path = input_file.path,
+    )
+
+def _zip_swift_arg(module_name, swift_identifier, input_file):
+    return "{module_name}.framework/Modules/{module_name}.swiftmodule/{swift_identifier}.{ext}={file_path}".format(
+        module_name = module_name,
+        swift_identifier = swift_identifier,
+        ext = input_file.extension,
+        file_path = input_file.path,
+    )
+
+def _swift_static_framework_impl(ctx):
+    module_name = ctx.attr.framework_name
+    fat_file = ctx.outputs.fat_file
+
+    input_archives = []
+    input_modules_docs = []
+    zip_args = [_zip_binary_arg(module_name, fat_file)]
+
+    for platform, archive in ctx.split_attr.archive.items():
+        swiftmodule_identifier = _PLATFORM_TO_SWIFTMODULE[platform]
+        if not swiftmodule_identifier:
+            fail("Unhandled platform '{}'".format(platform))
+
+        #library = archive[CcInfo].linking_context.libraries_to_link[0].pic_static_library
+        swift_info = archive[SwiftInfo]
+        swiftdoc = swift_info.direct_swiftdocs[0]
+        swiftmodule = swift_info.direct_swiftmodules[0]
+
+        libraries = archive[CcInfo].linking_context.libraries_to_link
+        archives = []
+        for library in libraries:
+            archive = library.pic_static_library or library.static_library
+            if archive:
+                archives.append(archive)
```

Comment:
I think we should add an `else` here and fail since everything should support this